### ISSUE TITLE
Question:  Possible way to create a custom work alike in cucumber-js?

### ISSUE THREAD

**Issue by zmimi-lael** [vitalets/playwright-bdd#132]

**Comment by zmimi-lael** [comment]
Goal is to have a this.context from a step file so it can share some information across different file steps?

**Comment by vitalets** [comment]
Yes, you can create `CustomWorld` inherited from `BddWorld`. [More details in docs.](https://vitalets.github.io/playwright-bdd/#/writing-steps/cucumber-style?id=custom-world)

**Comment by zmimi-lael** [comment]
@vitalets I tried to use the CustomWorld.

![image](https://github.com/vitalets/playwright-bdd/assets/148584709/3f613aa2-15bc-48e9-8818-e8aaa513fa70)

![image](https://github.com/vitalets/playwright-bdd/assets/148584709/4beb5f84-fb4a-43e3-b75d-4a8b550d3e26)

I am implementing my own `this._context` as a global storage. But, looks like I am missing something. I tried to use the `Given<CustomWorld>` from the example. Im using js btw.

**Comment by zmimi-lael** [comment]
@vitalets Hi, good day!

It worked! But not sure why :D
I do not add `CustomWorld` like this `Given<CustomWorld>. Since I am using `js`, I added the `**/*.js` in the config file and it worked.

![image](https://github.com/vitalets/playwright-bdd/assets/148584709/a2f865d0-db73-496b-af87-2345bf4860ff)
![image](https://github.com/vitalets/playwright-bdd/assets/148584709/976e073d-c23b-4af1-9bde-b9cb2149ea89)

**Comment by vitalets** [comment]
Hi @zmimi-lael ! You are right, `Given<CustomWorld>` is only for TypeScript, for js you can just write `Given(...)`

**Comment by zmimi-lael** [comment]
Thank you so much @vitalets ! Appreciate it.
